src/stash/merge.rs

```diff
@@ -0,0 +1,408 @@
+                        ..
+                    },
+                    OwnedState::ConfidentialAmount {
+                        seal_definition: seal,
+                        ..
+                    },
+                ) => {
+                    *self = OwnedState::Revealed {
+                        seal_definition: seal.clone(),
+                        assigned_state: state.clone(),
+                    };
+
+                    Ok(true)
+                }
+
+                // ConfidentialSeal + ConfidentialAmount = Revealed
+                (
+                    OwnedState::ConfidentialAmount {
+                        seal_definition: seal,
+                        ..
+                    },
+                    OwnedState::ConfidentialSeal {
+                        assigned_state: state,
+                        ..
+                    },
+                ) => {
+                    *self = OwnedState::Revealed {
+                        seal_definition: seal.clone(),
+                        assigned_state: state.clone(),
+                    };
+
+                    Ok(true)
+                }
+
+                // if self and other is of same variant do nothing
+                (
+                    OwnedState::ConfidentialAmount { .. },
+                    OwnedState::ConfidentialAmount { .. },
+                ) => Ok(true),
+                (
+                    OwnedState::ConfidentialSeal { .. },
+                    OwnedState::ConfidentialSeal { .. },
+                ) => Ok(true),
+
+                // Anything + Confidential = Anything
+                (_, OwnedState::Confidential { .. }) => Ok(true),
+
+                // Confidential + Anything = Anything
+                (OwnedState::Confidential { .. }, _) => {
+                    match other {
+                        OwnedState::ConfidentialSeal {
+                            seal_definition,
+                            assigned_state,
+                        } => {
+                            *self = OwnedState::ConfidentialSeal {
+                                seal_definition: seal_definition.clone(),
+                                assigned_state: assigned_state.clone(),
+                            };
+                            Ok(true)
+                        }
+
+                        OwnedState::ConfidentialAmount {
+                            seal_definition,
+                            assigned_state,
+                        } => {
+                            *self = OwnedState::ConfidentialAmount {
+                                seal_definition: seal_definition.clone(),
+                                assigned_state: assigned_state.clone(),
+                            };
+                            Ok(true)
+                        }
+
+                        // Other patterns are covered already. Should not reach
+                        // here
+                        _ => unreachable!(),
+                    }
```

Comment:
Just take other and return it back, in one line of code

Reply:
Updated.